### DISCUSSION TITLE
Graceful shutdown

### DISCUSSION THREAD

**Discussion by oblique** [libp2p/rust-libp2p#5416] [Q&A] [answered]
What is the best way to gracefully shutdown the `Swarm`?

**Comment by dariusc93** [comment]
The best way for now would be to drop `Swarm`. There isnt any way to gracefully shut it down directly though but dropping it should be enough and generally if its an application, you can just close the application.

EDIT: Just as a note, `Swarm` to my knowledge does not impl `Drop`

**Comment by mxinden** [comment, marked as answer]
Past discussion https://github.com/libp2p/rust-libp2p/pull/1682.
